do.sh

```diff
@@ -34,11 +34,39 @@ ovn_tester=${topdir}/ovn-tester
 EXTRA_OPTIMIZE=${EXTRA_OPTIMIZE:-no}
 USE_OVSDB_ETCD=${USE_OVSDB_ETCD:-no}
 
+DISTRO_IDS=$(awk -F= '/^ID/{print$2}' /etc/os-release)
+
+function is_rpm_based() {
+    case $DISTRO_IDS in
+    centos* | rhel* | fedora*)
+        true
+    ;;
+    *)
+        false
+    ;;
+    esac
+}
+
+function is_deb_based() {
+    case $DISTRO_IDS in
+    debian* | ubuntu*)
+        true
+        ;;
+    *)
+        false
+        ;;
+    esac
+}
+
 function die() {
     echo $1
     exit 1
 }
 
+function die_distro() {
+    die "Unable to determine distribution type."
```

Comment:
Should we say here that we currently only detect rpm-based and deb-based distros?

Reply:
:+1: